modules/internal/ChapelAutoAggregation.chpl

```diff
@@ -60,9 +60,50 @@ module ChapelAutoAggregation {
     return nil;  // return type signals that we shouldn't aggregate
   }
 
+  private proc isHeterogeneousTuple(type t) param {
+    if (isTuple(t) && !isHomogeneousTuple(t)) {
+      return true;
+    }
+    else {
+      return false;
+    }
+  }
+
+  private proc elemTypeSupportsAggregation(type t) param {
+    if (t == int || t == uint || t == real || t == bool) {
+      return true;
+    }
+    if (isHomogeneousTuple(t)) {
+      for param i in 0..<t.size {
+        if (!elemTypeSupportsAggregation(t[i])) {
+          return false;
+        }
+      }
+      return true;
+    }
+    if (isHeterogeneousTuple(t)) {
+      return elemTypeSupportsAggregation(t, 0);
+    }
+    
+    return false;
+  }
+
+  private proc elemTypeSupportsAggregation(type t, param dim: int) param
+      where isHeterogeneousTuple(t) {
+
+    if (dim+1 == t.size) {
+      return elemTypeSupportsAggregation(t[dim]);
+    }
+    else {
+      return elemTypeSupportsAggregation(t[dim]) &&
+             elemTypeSupportsAggregation(t, dim+1);
+
+    }
+  }
```

Comment:
I don't think you should do any of that. Just use `isPOD`.

``` chapel
private proc elemTypeSupportsAggregation(type t) param {
  return isPODType(t);
}
```

See also 

https://github.com/chapel-lang/chapel/blob/e7a41c2f9891ff292173425786d9082fe23b9f47/modules/standard/Types.chpl#L129-L151

Reply:
Yeah, should just use `isPOD`. I thought the unordered compiler optimization already included the pod check though.